### PR TITLE
Add markdown to document the database design

### PR DESCRIPTION
**Pull request by iambillmccann**
@dhrubilk @bruno-mota please review the markdown for the database design. Be sure to comment ... accept the PR when you are comfortable with the document.

### CODE REVIEW

**Comment by iambillmccann** [comment]
Added reviewers/assignees

**Review by dhruvilk — COMMENTED** [review]
Schema in the future may pose an issue given we are a bit confused on why the provider table is entirely necessary when we can simply assign providerIDs in the user table and query those. Would it be ok for us to do that? "Ignore" the provider table and just have it there for storage purposes?

**Comment by bruno-mota** [comment]
For authentication, how would we know if a user is connected to a provider if they were to log in using Gmail? 
Could we store the User table in Azure and use, let's say use firebase for authentication?
If the user's(doctor, nurse) email, from firebase, is in the table, then we can sign them in, else they will be logged out.

**Comment by iambillmccann** [comment]
> Schema in the future may pose an issue given we are a bit confused on why the provider table is entirely necessary when we can simply assign providerIDs in the user table and query those. Would it be ok for us to do that? "Ignore" the provider table and just have it there for storage purposes?

No. We cannot ignore the provider table. If it helps to understand better, don't think of the provider as a person. In fact, in many cases the provider will not be a person. Instead think of the provider as the doctor's business, or of a medical practice. 

You will, by the way, assign providerIDs in the user table. And we definitely cannot assign a unique providerID to each user. I thought I described this thoroughly. A doctor's office will have more than one user. The patients will be assigned to the office (provider) and not the user (doctor, nurse, or admin). 

Really, it's the user table that is less important. The user table only servers two purposes, one, for authentication and permission, and two to display a person's name when they are logged in.

File: api/db/database requirements.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+| id          | Int       | PK  | autoincrement           |
+| providerId  | Int       | FK  | required                |
+| active      | Boolean   |     | required, default=true  |
+| overdue     | Boolean   |     | required, default=false |
+| riskStatus  | Int       |     |                         |
+| firstName   | String    |     | required                |
+| middleName  | String    |     |                         |
+| lastName    | String    |     | required                |
+| avatar      | Blob      |     |                         |
+|             |           |     |                         |
```
Comment (dhruvilk) [inline api/db/database requirements.md:47]: add phone number as string and attribute would be required maybe but leave blank for now?

**Comment by iambillmccann** [comment]
> For authentication, how would we know if a user is connected to a provider if they were to log in using Gmail? Could we store the User table in Azure and use, let's say use firebase for authentication? If the user's(doctor, nurse) email, from firebase, is in the table, then we can sign them in, else they will be logged out.

There are a couple of questions here, so let me see if I can clarify them individually ...

- We will need a process for associating each user with the provider they work for. I brought this up with Artin and he indicated that Levl will register providers and users.
- Given the above, we will need separate pages for this purpose. At Glidian, we had a separate app for this purpose, and I may decide to do the same for us.
- And yes, we could (and will) store the user table separate from Firebase (or Auth0 or Google or whatever).
- If we use Firebase (or Auth0) then our user table would only keep the data that we cannot get from Firebase (or Auth0). That said, we would need an identifier that we can use to look up users in both our table and Firebase, that should be the only redundant user data.

**Comment by iambillmccann** [comment]
BTW. One of you will have to accept the PR in order for me to merge.

**Review by dhruvilk — APPROVED** [review]
PR Done

**Review by bruno-mota — APPROVED** [review]
Looks great for now! We will get to work on the mock data and authentication.

**Pull request merged**